src/animated/Controller.ts

```diff
@@ -4,61 +4,84 @@ import {
   is,
   toArray,
   withDefault,
+  hasKeys,
 } from '../shared/helpers'
 import AnimatedValue from './AnimatedValue'
 import AnimatedValueArray from './AnimatedValueArray'
 import { start, stop } from './FrameLoop'
 import { colorNames, interpolation as interp, now } from './Globals'
+import { SpringProps } from '../../types/renderprops'
 
-type FinishedCallback = (finished?: boolean) => void
+type Animation = any
+
+type ControllerProps<DS extends object> = DS &
+  SpringProps<DS> & {
+    attach?: (ctrl: Controller) => Animation
+  }
 
 type AnimationsFor<P> = { [Key in keyof P]: any }
 
 type ValuesFor<P> = { [Key in keyof P]: any }
 
-type InterpolationsFor<P> = {
-  [Key in keyof P]: P[Key] extends ArrayLike<any>
+type InterpolationsFor<DS> = {
```

Comment:
I'm not actually sure what `DS` stands for, but it's being used in the `SpringProps` type, so I wanted to match with that.

Reply:
@jacobrask @Jessidhia